### PR TITLE
AP-9819 # Changed calculation from element to use sdk-core

### PR DESCRIPTION
**Pull request by mymattcarroll**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

---

> [!NOTE]
> **Medium Risk**
> Shared calculation behavior now depends on sdk-core parity with the removed local logic; the simplified `onChange` effect may trigger extra updates compared to the previous equality guard.
> 
> **Overview**
> **Calculation elements** no longer evaluate expressions in `FormElementCalculation` with local `morph-expressions` setup. They call **`calculationService.evaluateExpression()`** from `@oneblink/sdk-core`, passing the calculation string, submission model, form elements, and `localisationService.parseDayOnlyDate`.
> 
> The in-component parser, `{ELEMENT:…}` property wiring, and custom functions (`ROUND`, `ISNULL`, etc.) are removed in favor of the shared SDK implementation. Direct app dependencies on **`morph-expressions`** and **`escape-string-regexp`** are dropped (expression parsing now comes via **`@oneblink/sdk-core` `10.1.0-beta.2`**), and the local **`morph-expressions` type declaration** is deleted.
> 
> The submission sync effect now always pushes the evaluated result through **`onChange`** (mapping `NaN` to `undefined`) instead of skipping updates when the numeric value is unchanged.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7138464000704e9947467f0a1e007e6bd05174b9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by divporter — APPROVED** [review]

File: src/form-elements/FormElementCalculation.tsx [hunk 1]
```diff
@@ -282,44 +68,18 @@ function FormElementCalculation({ element, onChange, value }: Props) {
...
-    } else {
-      onChange(element, {
-        value: undefined,
-      })
-    }
-  }, [element, formSubmissionModel, onChange, value, calculateContent])
+    onChange(element, {
+      value: calculatedValue,
+    })
+  }, [element, onChange, calculatedValue])
```
[the remark this answers is not in the document]
Comment (mymattcarroll) [reply src/form-elements/FormElementCalculation.tsx:79]: The unchange logic is handled by the `calculatedValue` being a dependency. If it does not change, the effect will not run.

I need to convert `NaN` to undefined. Fixing now.

File: src/form-elements/FormElementCalculation.tsx [hunk 2]
```diff
@@ -17,21 +15,6 @@ type Props = {
...
-  return !value && value !== 0
-}
-
-const isObjectWithValue = (obj: unknown): obj is { value: unknown } => {
-  return typeof obj === 'object' && obj !== null && 'value' in obj
-}
-
-// workaround for .toFixed() not rounding floating point numbers correctly
-function roundToFixed(number: number, decimals: number) {
-  const multiplier = Math.pow(10, decimals)
```
[the remark this answers is not in the document]
Comment (mymattcarroll) [reply src/form-elements/FormElementCalculation.tsx:30]: I will remove.

File: src/form-elements/FormElementCalculation.tsx [hunk 3]
```diff
@@ -55,223 +38,26 @@ function FormElementCalculation({ element, onChange, value }: Props) {
...
-      {
-        replacement,
-        nestedElementNames,
-      }: {
-        replacement: string
-        nestedElementNames: string[]
-      },
-    ) => {
-      exprParser.registerProperty(
-        replacement,
```
[the remark this answers is not in the document]
Comment (mymattcarroll) [reply src/form-elements/FormElementCalculation.tsx:74]: This is intended.
